src/main/kotlin/org/misarch/catalog/service/ProductService.kt

```diff
@@ -82,7 +82,8 @@ class ProductService(
             product.defaultVariantId = input.defaultVariantId
         }
         val savedProduct = repository.save(product).awaitSingle()
-        eventPublisher.publishEvent(CatalogEvents.PRODUCT_UPDATED, savedProduct.toEventDTO())
+        val categoryIds = productToCategoryRepository.findByProductId(savedProduct.id!!).map { it.categoryId }
+        eventPublisher.publishEvent(CatalogEvents.PRODUCT_UPDATED, savedProduct.toEventDTO(categoryIds.toSet()))
```

Comment:
It's a bit weird to switch between `Set` and `List` so often, but I guess it's okay.

Reply:
my reason is to get rid of duplicates (and the event wants a list afaik)
making a set temporarily out of it was the easiest way I could think of (and with the other functions more importantly the most LOC efficient way)